patches/node/fix_ensure_traverseparent_bails_on_resource_path_exit.patch

```diff
@@ -0,0 +1,67 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Shelley Vohr <shelley.vohr@gmail.com>
+Date: Tue, 18 Mar 2025 10:41:27 +0100
+Subject: fix: ensure TraverseParent bails on resource path exit
+
+Electron's security model requires that we do not traverse outside of the
+resource path. This commit ensures that the TraverseParent function
+bails out if the parent path is outside of the resource path.
+
+diff --git a/src/node_modules.cc b/src/node_modules.cc
+index 210a01d24e11764dc9fc37a77b354f11383693f8..8257f8eb057a374defd19b19d6056e01c5b9b30b 100644
+--- a/src/node_modules.cc
++++ b/src/node_modules.cc
+@@ -290,8 +290,40 @@ const BindingData::PackageConfig* BindingData::TraverseParent(
+     Realm* realm, const std::filesystem::path& check_path) {
+   std::filesystem::path current_path = check_path;
+   auto env = realm->env();
++  Isolate* isolate = env->isolate();
+   const bool is_permissions_enabled = env->permission()->enabled();
+ 
++  // Get the resources path with trailing slash.
++  std::string resources_path;
++  {
++    HandleScope handle_scope(isolate);
++    Local<Value> resources_path_value;
++    Local<Value> process_type_value;
++    bool is_browser_process = false;
++    Local<Object> process_object = env->process_object();
++
++    Local<String> type_path_key =
++        String::NewFromUtf8(isolate, "type").ToLocalChecked();
++    if (process_object->Get(env->context(), type_path_key)
++            .ToLocal(&process_type_value) &&
++        process_type_value->IsString()) {
++      String::Utf8Value type_str(isolate, process_type_value);
++      is_browser_process = (strcmp(*type_str, "browser") == 0);
++    }
++
++    Local<String> resources_path_key =
++        String::NewFromUtf8(isolate, "resourcesPath").ToLocalChecked();
++    if (is_browser_process &&
++        process_object->Get(env->context(), resources_path_key)
++            .ToLocal(&resources_path_value) &&
++        resources_path_value->IsString()) {
++      resources_path = *String::Utf8Value(isolate, resources_path_value);
++      if (!resources_path.empty() && resources_path.back() != kPathSeparator) {
++        resources_path += kPathSeparator;
++      }
++    }
++  }
++
+   do {
+     current_path = current_path.parent_path();
+ 
+@@ -310,6 +342,12 @@ const BindingData::PackageConfig* BindingData::TraverseParent(
+       return nullptr;
+     }
+ 
++    // If current path is outside the resources path, bail.
++    if (!resources_path.empty() &&
++        !current_path.generic_string().starts_with(resources_path)) {
```

Comment:
I think `starts_with` should be replaced with a path util thing here, case insensitive file systems and all 😅